rtl/eclrtl/eclregex.cpp

```diff
@@ -246,26 +266,27 @@ class CCompiledStrRegExpr : implements ICompiledStrRegExpr
         uint32_t options = ((_isCaseSensitive ? 0 : PCRE2_CASELESS) | (_enableUTF8 ? PCRE2_UTF : 0));
         size32_t regexSize = (isUTF8Enabled ? rtlUtf8Size(_regexLength, _regex) : _regexLength);
 
-        compiledRegex = pcre2_compile_8((PCRE2_SPTR8)_regex, regexSize, options, &errNum, &errOffset, pcre2CompileContext8);
+        pcre2_code_8 * newCompiledRegex = pcre2_compile_8((PCRE2_SPTR8)_regex, regexSize, options, &errNum, &errOffset, pcre2CompileContext8);
 
-        if (compiledRegex == nullptr)
+        if (newCompiledRegex == nullptr)
         {
-            failWithPCRE2Error(errNum, "Error in regex pattern: ", _regex, errOffset);
+            failWithPCRE2Error(errNum, "Error in regex pattern: ", std::string(_regex, _regexLength), errOffset);
         }
-    }
 
-    ~CCompiledStrRegExpr() //CAVEAT non-virtual destructor !
-    {
-        pcre2_code_free_8(compiledRegex);
+        compiledRegex = std::shared_ptr<pcre2_code_8>(newCompiledRegex, pcre2_code_free_8);
     }
 
+    CCompiledStrRegExpr(CCompiledStrRegExpr & other) = default; // Note non-const argument
```

Comment:
Why the need for the non-const argument?

Reply:
The std::shared_ptr in other apparently needs to be updated (the compiler complained about the original const version I had).  If you know of a pattern where I can make other const, please tell me!

Reply:
Ah, never mind.  I think I had a different underlying issue at the time.  All copy constructors now accept const arguments.